e-commerce/shop-backend/src/operations.ts

```diff
@@ -264,6 +267,14 @@ export class Shop {
 
   @Communicator()
   static async createPaymentSession(ctxt: CommunicatorContext, productDetails: Product[], origin: string): Promise<PaymentSession> {
+    if (origin === 'xxx') {
+        // TODO: Only in test mode
+        return {
+            session_id: "1234",
+            url:ctxt.workflowUUID,
+            payment_status: "pending",
+        };
+    }
     const { paymentHost, localHost } = getHostConfig(ctxt);
 
     const response = await fetch(`${paymentHost}/api/create_payment_session`, {
```

Comment:
If we use `spyOn` for this `fetch` call, maybe we don't need to do the `origin == 'xxx'` trick above.

Reply:
I have a better idea, I think.

Reply:
It is unfortunate that spyOn doesn't work in a straightforward manner on our code.  I worked around it in a less brutal way than stubbing out fetch entirely... and in a way that would be illustrative of more complex scenarios.